docroot/sites/facilities.uiowa.edu/modules/facilities_core/src/BuildingsProcessor.php

```diff
@@ -65,6 +104,52 @@ protected function processRecord(&$record) {
     }
   }
 
+  /**
+   * Save a local version of an image gotten from the facilities API
+   *     and assign that as the building image.
+   *
+   * @param array $result
+   *   The result array reference that contains the image URL.
+   *
+   * @return null
+   *   This function modifies the image URL in place on the result reference.
+   *
+   * @throws GuzzleException
+   *    Throws an exception if it can't get the image for the building.
+   */
+  protected function processResultImage(&$result) {
+    $this->client = \Drupal::service('http_client');
+    $this->fs = \Drupal::service('file_system');
+    $this->configFactory = \Drupal::service('config.factory');
+    $this->imageFieldConfig = FieldConfig::loadByName('node', 'building', 'field_building_image');
```

Comment:
We still need to put this outside the foreach loop. I have an idea how to do it.

Reply:
I made an init function for these. If you have another idea, let me know.